src/Analysis/GWASResults/Views/Home/InitialState/DetermineInitialColumnManagement.js

```diff
@@ -0,0 +1,45 @@
+import localStorageAvailable from './localStorageAvailable';
+import DefaultColumnManagement from './DefaultColumnManagement';
+
+const hasValidKeys = (a, b) => {
+  // Checks that the keys of two objects match
+  const aKeys = Object.keys(a).sort();
+  const bKeys = Object.keys(b).sort();
+  return JSON.stringify(aKeys) === JSON.stringify(bKeys);
+};
+
+const hasOnlyBoolValues = (obj) => Object.values(obj).every(Boolean);
+
+const columnManagementLocalStorageIsValid = () => {
+  const retrievedObject = localStorage.getItem('columnManagement');
+  const parsedRetrievedObject = JSON.parse(retrievedObject);
+  return (
+    hasValidKeys(parsedRetrievedObject, DefaultColumnManagement)
+    && hasOnlyBoolValues(retrievedObject)
+  );
+};
+
+const DetermineInitialColumnManagement = () => {
+  if (localStorageAvailable()) {
```

Comment:
Note: I found out while testing this that the underlying data portal application requires localstorage to work. Otherwsie the entire application crashes. This is due to line 180 in data-portal/src/DataDictionary/utils.js

```
/**
 * Get search history items from localStorage
 * @returns {SearchHistoryItem[]} array of search history items
 */
export const getSearchHistoryItems = () => {
  const items = JSON.parse(localStorage.getItem(searchHistoryLocalStorageKey));
  return items;
};
```
We might want to revisit this function if we plan on supporting users with localstorage disabled.

Reply:
your solution looks good to me 👍